src/tools/scripts/deploy.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# exit when any command fails instead of trying to continue on
+set -e
+
+echo "Update local production branch"
+git checkout production
+git status
+git pull
+git pull origin main
+
+if [ "$(pgrep -f 'python main.py')" ]; then
+  echo "Killing existing server to run afresh"
+  pkill -9 python main.py
+fi
+
+echo "Run and test website"
+./tools/scripts/run_and_test_website.sh
+
+echo "Generate ebooks"
+npm run ebooks
```

Comment:
What's the behavior if the deployer doesn't have the ebook prerequisites (Prince installed)? Does the deploy fail or does it proceed without ebooks?

Reply:
It fails and exits instead of continuing. Prince is now basically a requirement to run this script. We can run individual steps manually if really want to avoid it.

I’m in two kinds here too.

The “Generate Ebooks” GitHub action works quite well and avoid having to do local installs with Git. But at the expense of having to remember to run that first and store the results in Git.

Ultimately I see this as a short term solution and I’d like to love to CI/CD so adding burden of installing PrinceXML for the few deployers we have is OK with me.

Or alternatively we keep PDFs in Git and only regenerate them periodically via the GitHub action if prince not installed?

Reply:
OK change of plan here. I've reverted the change so eBooks are still stored in Git. This allows us to run the GitHub action to generate the ebooks. I've added a step to confirm this is done:

```bash
read -r -n 1 -p "Please confirm you've updated the eBooks via GitHub Actions. [y/N]: " DEPLOY
if [ "$DEPLOY" != "Y" ] && [ "$DEPLOY" != "y" ]; then
    echo
    echo "Cancelling deploy"
    exit 0
fi
echo
```